src/deadline/client/ui/widgets/host_requirements_tab.py

```diff
@@ -431,6 +506,39 @@ def get_requirements(self) -> Dict[str, List]:
 
         return requirements
 
+    def set_requirements(self, requirements: Dict[str, List]):
+        amount_widgets = {}
+        attribute_widgets = {}
+        for i in range(self.list_widget.count()):
+            widget = self.list_widget.itemWidget(self.list_widget.item(i))
+            widget_requirement = widget.get_requirement()
+            if widget_requirement:
```

Comment:
Similar to my other comment, I think we've generally got the right idea here, but there are some small updates needed.

I think our desired behaviour should be:
1. If there is an input custom requirement, we should have a custom widget for it, even if it doesn't already exist in the UI
2. If there is a custom requirement in the UI that isn't in the input, then we should remove the widget for it.

Based on my understanding of this code, if a widget does not already exist for a custom capability (which would be the case for all custom capabilities when the host requirements are first created), then we'll just get a warning and no widget will be created, right? And vice versa - if there is a widget that already exists for a custom attribute, it won't be deleted, even if it's not in the input.

Thoughts?